src/domains/FeedbackPortal/ReportHistory/ReportHistory.tsx

```diff
@@ -0,0 +1,267 @@
+    getBugReportsBySubmitter,
+} from '../api';
+
+import { FeedbackReport, BugReport } from '../types';
+
+const ReportOptions = ['Feedback', 'Bug'];
+
+const sortingOptions = [
+    { name: 'Ascending', value: 'true' },
+    { name: 'Descending', value: 'false' },
+];
+
+type Report = FeedbackReport | BugReport;
+
+// TODO: auth
+const user = {
+    _id: '123456789',
+};
+
+const pageSize = 10;
+
+const useStyles = makeStyles((theme: Theme) => ({
+    select: {
+        borderColor: theme.palette.common.white,
+        color: theme.palette.common.white,
+    },
+}));
+
+export default function ReportHistory() {
+    const classes = useStyles();
+    const [prevReportType, setPrevReportType] = React.useState('');
+    const [reportType, setReportType] = React.useState('');
+    const [sortingOrder, setSortingOrder] = React.useState('');
+    const [page, setPage] = React.useState(1);
+    const [numOfPages, setNumOfPages] = React.useState(0);
+    const [reports, setReports] = React.useState<Report[]>([]);
+
+    const handleReportChange = (e: React.ChangeEvent<{ value: unknown }>) => {
+        setReportType(e.target.value as string);
+    };
+
+    const handleSortingChange = (e: React.ChangeEvent<{ value: unknown }>) => {
+        setSortingOrder(e.target.value as string);
+    };
+
+    const feedbackReportsAPIrequest = React.useCallback(
+        () =>
+            // TODO: Replace with user Id
+            getFeedbackReportsBySubmitter(page, sortingOrder, user._id),
+        [page, sortingOrder]
+    );
+
+    const bugReportsAPIrequest = React.useCallback(
+        // TODO: Replace with user Id
+        () => getBugReportsBySubmitter(page, sortingOrder, user._id),
+        [page, sortingOrder]
+    );
+
+    const [sendFeedbackRequest, isLoadingFeedback] = useEndpoint(
+        feedbackReportsAPIrequest,
+        {
+            onSuccess: (results) => {
+                // Adds type attribute to report objects. This will be needed in children components
+                const feedbackReports = results.data.reports.map((report) => ({
+                    ...report,
+                    type: 'Feedback',
+                })) as Report[];
+                setNumOfPages(results.data.count / pageSize);
+                setReports(feedbackReports);
+            },
+        }
+    );
+
+    const [sendBugRequest, isLoadingBug] = useEndpoint(bugReportsAPIrequest, {
+        onSuccess: (results) => {
+            const bugReports = results.data.reports.map((report) => ({
+                ...report,
+                type: 'Bug',
+            })) as Report[];
+            setNumOfPages(results.data.count / pageSize);
+            setReports(bugReports);
+        },
+    });
+
+    const sendRequest = () => {
+        // Clean reports from state of component
+        setReports([]);
+        // If the report type selected has changed then set the page number to 1
+        if (prevReportType !== reportType) {
+            setPage(1);
+        }
```

Comment:
Out of curiosity, what's the reasoning for clearing the reports and setting the # of pages to 1 right before we send the request?

Reply:
I clear the reports so that the bottom pagination does not show while the request is being sent (looks kind of weird when both the pagination and loader are rendered at the same time). 

The page number is set to 1 only if the user changes the report type selected. For example, user is currently looking at feedback reports and is on page 8. Let's say now user selects bug reports and clicks the search button. A request to the bug reports API would be sent with query parameters sortByDate=true and page=8. However, we don't know if there are 8 pages of bug reports. That's why I set to 1.

Reply:
I think we agreed in our last meeting that this will be resolved via infinite scrolling in your next PR, so we will leave this as is for now.